modules/qna/src/backend/storage.ts

```diff
@@ -142,12 +143,15 @@ export default class Storage {
     const qnaItems = (await this.fetchQNAs()).filter(q => !editingQnaId || q.id != editingQnaId)
 
     const newQuestions = Object.values(newItem.questions).reduce((a, b) => a.concat(b), [])
-    const dupes = _.flatMap(qnaItems, item => Object.values(item.data.questions))
-      .reduce((a, b) => a.concat(b), [])
-      .filter(q => newQuestions.includes(q))
+    const dupes = qnaItems
+      .map(item => ({
+        id: item.id,
+        questions: Object.values(item.data.questions).reduce((acc, arr) => [...acc, ...arr], [])
+      }))
+      .filter(existingQuestion => !!existingQuestion.questions.filter(q => newQuestions.includes(q)).length)
 
     if (dupes.length) {
-      throw new Error(`These questions already exist in another entry: ${dupes.join(', ')}`)
+      throw new Error(JSON.stringify({ error: 'duplicated_question', items: dupes.map(item => item.id) }))
```

Comment:
I don't like errors being used like that. You could either add an API method to check duplicates, or if you want to use a single call, then either return `status(200)` for a successful insertion, or `status(400).send({ duplicated_questions... })` instead (or some other status code)

You won't have to handle the stringify/parse.

Also, you can create custom errors, for example you could throw a DuplicateEntries(items), then in the API you wrap the call in a try catch, then you can check if `err instanceof DuplicateEntries` and you can return the custom payload

Reply:
I decided to just revert it as I'd rather focus on the front-end for now, I just changed the message to pass a short string easy to identify in the front-end